packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -805,6 +812,7 @@ export class Function extends FunctionBase {
       fileSystemConfigs,
       codeSigningConfigArn: props.codeSigningConfig?.codeSigningConfigArn,
       architectures: this._architecture ? [this._architecture.name] : undefined,
+      snapStart: props.snapStart ? { applyOn: 'PublishedVersions' } : undefined,
```

Comment:
I love the clean design but I am also wary if they will support all versions including the $latest one. But I agree we keep the simple design as is.

Reply:
Yes, we should. Resource creation on AWS throws an error as there is a validation that they do. So it makes sense to add the validation to CDK build. Added validation, and provided unit test for the same.